lib/avro_turf/messaging.rb

```diff
@@ -120,13 +120,19 @@ def initialize(
     # validate    - The boolean for performing complete message validation before
     #               encoding it, Avro::SchemaValidator::ValidationError with
     #               a descriptive message will be raised in case of invalid message.
+    # read_only   - The boolean that indicates whether or not the schema should be
+    #               registered in case it does not exist, or if it should be fetched
+    #               from the registry without registering it (read_only: true).
     #
     # Returns the encoded data as a String.
-    def encode(message, schema_name: nil, namespace: @namespace, subject: nil, version: nil, schema_id: nil, validate: false)
+    def encode(message, schema_name: nil, namespace: @namespace, subject: nil, version: nil, schema_id: nil, validate: false,
+               read_only: false)
```

Comment:
I think `read_only` is a bit confusing as the name for this – the side effect of registering a schema is not the _primary_ job of `encode`, so `read_only` would seem to apply to the encoding of data.

I think `register_schemas` makes more sense?

Reply:
Good point! changing it 👍